### ISSUE TITLE
[🐛 Bug Report] Some images in val/test sets are filtered by program 

### ISSUE THREAD

**Issue by Scarecrow0** [KaihuaTang/Scene-Graph-Benchmark.pytorch#35]
## 🐛 Bug
Hi,
I found a little glitch in your code.
At the config for build up the datasets [maskrcnn_benchmark/config/paths_catalog.py, line 159](https://github.com/KaihuaTang/Scene-Graph-Benchmark.pytorch/blob/master/maskrcnn_benchmark/config/paths_catalog.py#L159).

``` python
# IF MODEL.RELATION_ON is True, filter images with empty rels
# else set filter to False, because we need all images for pretraining detector
args['filter_non_overlap'] = (
                                 not cfg.MODEL.ROI_RELATION_HEAD.USE_GT_BOX) and cfg.MODEL.RELATION_ON and cfg.MODEL.ROI_RELATION_HEAD.REQUIRE_BOX_OVERLAP
args['filter_empty_rels'] = cfg.MODEL.RELATION_ON
args['flip_aug'] = cfg.MODEL.FLIP_AUG
return dict(
    factory="VGDataset",
    args=args,
)
```
The args 'filter_empty_rels' will be always set as True if the relation mode is set on, even while building up the dataset for validation or testing, So there are some images that are filtered in the test and validation set.
I tried to fix it by modifying the line159 to:
``` python
args['filter_empty_rels'] = cfg.MODEL.RELATION_ON and split == 'train'
```
The model(Motifs baseline) obtains a lower performance of object detection (mAP50 drop 4\~5) and scene graph generation(Recall@100 drop 4\~5 ). You may need to check this issue.

**Comment by Scarecrow0** [comment]
It seems that motifs codebase also filters the images of the test and validation set....
Sorry about that. Maybe it is a general preprocess setting.
